react/store/createReturnRequest/components/ItemsList.tsx

```diff
@@ -14,49 +16,100 @@ interface Props {
+    id: 'store/return-app.return-order-details.table-header.reason',
+  },
+  condition: {
+    id: 'store/return-app.return-order-details.table-header.condition',
+  },
+})
+
+const TableHeaderRenderer = (
+  formatMessage: IntlFormatters['formatMessage'],
+  addCondition: boolean
+) => {
+  return function Header(value: string) {
+    if (!addCondition && value === 'condition') {
+      return
+    }
+
+    return (
+      <th className="v-mid pv0 tl bb b--muted-4 normal bg-base bt ph3 z1 pv3-s">
+        {formatMessage(messages[value])}
+      </th>
+    )
+  }
+}
+
 export const ItemsList = (props: Props) => {
   const { items, creationDate } = props
 
   const { data: storeSettings } = useStoreSettings()
   const { options } = storeSettings ?? {}
   const { enableSelectItemCondition } = options ?? {}
 
+  const {
+    hints: { phone },
+  } = useRuntime()
+
+  const { formatMessage } = useIntl()
+
   const handles = useCssHandles(CSS_HANDLES)
   const { inputErrors } = useReturnRequest()
 
   const noItemSelected = inputErrors.some(
     (error) => error === 'no-item-selected'
   )
 
+  const TableHeader = TableHeaderRenderer(
+    formatMessage,
+    Boolean(enableSelectItemCondition)
+  )
+
   return (
-    <table className={`${handles.itemsListContainer} w-100`}>
+    <table
+      className={`${handles.itemsListContainer} w-100`}
+      style={{ borderCollapse: 'collapse' }}
+    >
       <thead
         className={`${handles.itemsListContainer} w-100 ph4 truncate overflow-x-hidden c-muted-2 f6`}
       >
         <tr className="w-100 truncate overflow-x-hidden">
-          <th className="v-mid pv0 tl bb b--muted-4 normal bg-base bt ph3 z1 pv3-s">
-            <FormattedMessage id="store/return-app.return-order-details.table-header.product" />
-          </th>
-          <th className="v-mid pv0 tl bb b--muted-4 normal bg-base bt ph3 z1 pv3-s tc">
-            <FormattedMessage id="store/return-app.return-order-details.table-header.quantity" />
-          </th>
-          <th className="v-mid pv0 tl bb b--muted-4 normal bg-base bt ph3 z1 pv3-s tc">
-            <FormattedMessage id="store/return-app.return-order-details.table-header.available-to-return" />
-          </th>
-          <th className="v-mid pv0 tl bb b--muted-4 normal bg-base bt ph3 z1 pv3-s tc">
-            <FormattedMessage id="store/return-app.return-order-details.table-header.quantity-to-return" />
-          </th>
-          <th className="v-mid pv0 tl bb b--muted-4 normal bg-base bt ph3 z1 pv3-s tc">
-            <FormattedMessage id="store/return-app.return-order-details.table-header.reason" />
-          </th>
-          {!enableSelectItemCondition ? null : (
-            <th className="v-mid pv0 tl bb b--muted-4 normal bg-base bt ph3 z1 pv3-s tc">
-              <FormattedMessage id="store/return-app.return-order-details.table-header.condition" />
-            </th>
-          )}
+          {phone ? mobileOrder.map(TableHeader) : desktopOrder.map(TableHeader)}
```

Comment:
I love the pattern here, but I would make it more explicit to avoid [this case of memory leak](https://twitter.com/andrestaltz/status/1473669713490235397) if someone changes the return of `TableHeaderRenderer`

```suggestion
          {phone ? mobileOrder.map((header) => TableHeader(header)) : desktopOrder.map((header) => TableHeader(header))}
```

Reply:
Added